tests/upgrades/test_hostcontent.py

```diff
@@ -0,0 +1,112 @@
+"""Test Hosts Content related Upgrade Scenarios
+
+:Requirement: UpgradedSatellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Host-Content
+
+:Assignee: spusater
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+from broker.broker import VMBroker
+from upgrade_tests.helpers.scenarios import create_dict
+from upgrade_tests.helpers.scenarios import get_entity_data
+
+from robottelo.hosts import ContentHost
+
+
+class TestScenarioDBseedHostMismatch:
+    """This test scenenario veryfies that the upgrade succeeds even when inconsistencies exist
+    in the database between Organization, Location and Content Host.
+
+    Test Steps:
+
+        1. Before Satellite upgrade
+        2. Create a New Organization and Location
+        3. Create a Content Host in the Organization
+        4. Ensure the Org is not in the Location
+        4. Assign the Content Host to the Location using the rake console
+        7. Ensure the Host is in both but the Org is not in Location, creating a mismatch
+        8. Upgrade Satellite
+        9. Ensure upgrade succeeds
+
+    BZ: 2043705, 2028786, 2019467
+    """
+
+    @pytest.mark.pre_upgrade
+    def test_pre_db_seed_host_mismatch(self, target_sat):
+        """
+        :id: 28861b9f-8abd-4efc-bfd5-40b7e825a941
+
+        :steps:
+            1. Create a Location
+            2. Create an Org and ensure the Org is not in the Location
+            3. Create a Content Host on Org
+            4. Use rake console to assign the Content Host to the Location
+            5. Ensure the mismatch is created for Content Host when Org is not in the Location
+            6. Do the upgrade
+
+        :expectedresults:
+            1. The Content Host is assigned to both Location and Org, but Org is not in Location
+
+        :BZ: 2043705, 2028786, 2019467
+
+        :customerscenario: true
+        """
+        org = target_sat.api.Organization().create()
+        loc = target_sat.api.Location().create()
+
+        chost_vm = VMBroker(nick='rhel7', host_classes={'host': ContentHost}).checkout()
```

Comment:
I think we are no more using `nick` based checkouts since we are directly passing argumets to parameters. Check other tests in robottelo.

Reply:
You can also use `rhel7_contenthost` global content host fixture which is not checking in the 
 content host!

Reply:
Hi, I can't seem to find any examples in robottelo that use VMBroker without `nick`, can you elaborate on how I can address this? 

I think we had a discussion about using `rhel7_contenthost` in the automation channel the other day, but perhaps I misunderstood. As I understood it `rhel7_contenthost` is session scoped and would not be available to the `post` scenario since `pre` and `post` are run in separate sessions. I think you had recommended calling broker explicitly to `checkout()` the vm in pre and then to `checkin()` at the end of post, but if there is a better way to handle this situation please let me know and I can fix it

Reply:
@spusateri Doesn't matters if it's a session-scoped but the good part is it doesn't have teardown after its yield also the upgrade scenario job is not `checking_in` all after pre-upgrade tests execution. You can try executing the test with `rhel7_contenthost` locally as well.

So you should be good to use `rhel7_contenthost`.

Fo what explicit parameters we use now with VMBroker, you can see the conf directory keys that we are returning in this def https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/core/contenthosts.py#L16